### DISCUSSION TITLE
v5.0.0-alpha1

### DISCUSSION THREAD

**Discussion by phaumer** [IBM/zopeneditor-about#447] [Announcements]
This is an alpha-level pre-release for Z Open Editor 5.0.0 that works with Zowe Explorer 3.0.0. It is not a feature-complete version of v5, not tested, and not documented. Use at your own risk to evaluate the Zowe v3 user experience and to report issues that still need fixing.

If you report a bug, please mention the build id `5.0.0-202409301253-d546c4f2` as we might publish more updates like these if we find and fix more issues.

Here are the still incomplete release notes for this version:

Breaking changes:

- Java 17: We are now building our language server with Java 17, which now the minimum Java version required for Z Open Editor. The editor will not work with Java 11 anymore. We recommend IBM's Semeru Runtime or JDK. Distributions from Oracle and OpenJDK will also work.
- VS Code v1.91 or newer: As part of the upgrade to Java 17 we are able to use newer versions of language server APIs provided by the Eclipse LSP4J project and Microsoft. As a result we also increase the required version for VS Code for our extension to v1.91 or newer.
- Zowe v3: Zowe Explorer 3.0.0 or newer is now required. Z Open Editor v5 will now download and install Zowe Explorer v3. Zowe Explorer v2 will not work anymore. For IBM RSE API Plugin for Zowe CLI the requirement is now Zowe CLI v8.

New features and fixes:

- Localization: Z Open Editor and Zowe Explorer are now available for the [VS Code display languages](https://code.visualstudio.com/docs/getstarted/locales) French, German, Japanese, Portuguese, and Spanish. This also includes syntax error messages provided by the COBOL, PL/I, JCL and REXX language servers. Download the respective language pack and switch.
- Added user settings to enable/disable folding range support for JCL and COBOL.

<hr /><em>This discussion was created from the release <a href='https://github.com/IBM/zopeneditor-about/releases/tag/5.0.0-alpha1'>v5.0.0-alpha1</a>.</em>

**Comment by FALLAI-Denis** [comment]
Hi,

For the Portuguese localization, you targeted the Brazilian variant: bundle.l10n.pt-br.json.
We have a population of Portuguese developers, and in our own extensions we used the base Portuguese, i.e. bundle.l10n.pt.json files.
Is there a reason to have selected the Brazilian variant of Portuguese?
But I see that for VS Code itself, there is only a Portuguese-Brazilian localization extension...
So that explains it...
We will also align ourselves with the Portuguese-Brazilian.

Thanks.

**Comment by phaumer** [comment]
Yes, but independently from that we selected the current list of language based on usage looking at the telemetry data that VS Code is collecting for an extension. Brazil is actually the second most popular country for our editor.

**Comment by phaumer** [comment]
One known issue is that the new Java 17 check did not make it into this build. So if you try to run it with Java 11 the Welcome page will say that it found it, but the language server will keep crashing. So ensure you have 17 specified as `JAVA_HOME` or the `"zopeneditor.JAVA_HOME"` setting.

**Comment by JillieBeanSim** [comment]
Known issue with RSE profile used in Zowe Explorer with this build when changing authentication method or stored values, an error is returned with `Failed to load schema for profile type rse`.

**Comment by FALLAI-Denis** [comment]
Hi,

I thought Zowe Explorer would handle files downloaded from z/OS or USS in memory instead of on disk.
I thought Z Open Editor would do the same, but I see that ZOE continues to store copybooks / includes on disk.
Is this the expected behavior in ZOE version 5.0.0?

Thanks.

**Reply by phaumer** [reply]
We intend on supporting this as soon as we can, but are running out of time for 5.0.0.

**Comment by phaumer** [comment]
Please, do not use this alpha anymore as the full release is now available.
